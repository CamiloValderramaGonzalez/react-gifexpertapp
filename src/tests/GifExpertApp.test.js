import React from 'react'
import { shallow } from 'enzyme'

import { GifExpertApp } from '../GifExpertApp'

describe('Pruebas en <GifExpertApp/>', () => {
    test('Debe ser igual al snapshot', () => {
        const wrapper = shallow(<GifExpertApp />)

        expect(wrapper).toMatchSnapshot();
    })

    test('Debe tener la misma cantidad de categorias que se envian', () => {
        const categories = ['One Puch', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
})