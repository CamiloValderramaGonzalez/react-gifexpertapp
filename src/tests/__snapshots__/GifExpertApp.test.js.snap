// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp/> Debe ser igual al snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en <GifExpertApp/> Debe tener la misma cantidad de categorias que se envian 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Puch"
      key="One Puch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
